src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,40 +1,66 @@
 import React from 'react';
 
+import { useAppSelector } from '../../app/hooks';
+import { useDispatch } from 'react-redux';
+import { filterSlice } from '../../features/filter';
+import { Status } from '../../types/Status';
+
 export const TodoFilter: React.FC = () => {
+  const filter = useAppSelector(state => state.filter);
+  const query = filter.query;
+  const dispatch = useDispatch();
+
+  const handleSelect = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    dispatch(filterSlice.actions.setStatus(event.target.value));
+  };
+
+  const ChangeQuery = (event: React.ChangeEvent<HTMLInputElement>) => {
+    dispatch(filterSlice.actions.setQuery(event.target.value));
+  };
```

Comment:
The function name `ChangeQuery` does not follow the camelCase convention for function names in JavaScript. It would be better to rename it to `changeQuery` to follow the convention and improve code consistency.